paasta_tools/docker_wrapper.py

```diff
@@ -65,15 +73,41 @@ def can_add_hostname(args):
             if 'h' in arg:
                 return False
 
+    return True
+
+
+def is_network_host(args):
+    for index, arg in enumerate(args):
         # Check for --network=host and variants
         if arg in ('--net=host', '--network=host'):
-            return False
+            return True
         try:
             if arg in ('--net', '--network') and args[index + 1] == 'host':
-                return False
+                return True
         except IndexError:
             pass
 
+    return False
+
+
+def is_run(args):
+    try:
+        list(args).index('run')
+        return True
+    except ValueError:
+        return False
```

Comment:
Yeah its true, though it's also an existing problem. Our arg parsing is not sophisticated enough to differentiate between which args are for top-level docker args vs the ones that get passed to individual commands.